lightblue/src/main/java/org/esbtools/eventhandler/lightblue/LightblueDocumentEventRepository.java

```diff
@@ -153,18 +148,6 @@ public void markDocumentEventsProcessedOrFailed(
         lightblue.bulkData(markDocumentEvents);
     }
 
-    private void blockUntilLockAcquired(String resourceId) throws LightblueException {
-        while (!locking.acquire(resourceId)) {
-            try {
-                // TODO: Parameterize lock polling interval
-                // Or can we do lock call that only returns once lock is available?
-                Thread.sleep(3000L);
-            } catch (InterruptedException e) {
-                // Just try again...
-            }
-        }
-    }
-
     /**
      * Creates {@link LightblueDocumentEvent}s from entities using
      * {@link #documentEventFactoriesByType}, optimizes away superseded and merge-able events, and
```

Comment:
Q: how useful is the parseAndOptimizeDocumentEventEntitiesUpTo()? It relies on the related events being in the same batch.